### ISSUE TITLE
ImageField undefined default connection in 0.8

### ISSUE THREAD

**Issue by vtable** [MongoEngine/flask-mongoengine#282]
test code : 

```python
from flask import Flask
from flask_mongoengine import MongoEngine
import StringIO

app = Flask(__name__)
db = MongoEngine()
db.init_app(app)

class Test(db.Document):
    name = db.StringField()
    image = db.ImageField()

@app.route('/db_init')
def db_init():
    t = Test(name='name1')
    t.save()

    return 'db init'

@app.route('/test1')
def test1():
    t = Test.objects.first()

    image_data = file('test.jpg', 'rb')
    t.image.put(image_data) # error

    return 'test1'

@app.route('/test2')
def test2():
    t = Test.objects.first()
    t.delete() # error
    # Test.objects.filter(name='name1').delete() # works fine
    return 'test2'

app.run(debug=True, host='127.0.0.1')
```
I get the error ("You have not defined a default connection") in two cases :

1. When putting data  with `t.image.put()` in a view.
2. When trying to delete Test model with ImageField  with`t.delete()` 
(however `Test.objects.filter(name='name1').delete()` works fine)

At first I thought this might be the result of #241 and indeed when downgrading to 0.7.5 the issues resolved but it can be relating also to #257.

**Comment by wojcikstefan** [comment]
Will probably be addressed by #280. @vtable can you try running this with the latest commit from that branch?

**Comment by vtable** [comment]
Yes, just checked it and it fixed the issue.
